policies/recipes/recipedb.go

```diff
@@ -89,8 +89,10 @@ func (db RecipeDB) addRecipe(name, version string, success bool) error {
 	if err != nil {
 		return err
 	}
-	defer f.Close()
-	if _, err = f.Write(dbBytes); err != nil {
+
+	_, err = f.Write(dbBytes)
+	f.Close()
```

Comment:
the file was still open when the os.rename was being called, which was causing errors on windows

Reply:
You need to check the return on the close:

if _, err = f.Write(dbBytes); err != nil {
f.Close()
return err
}

if err := f.Close() {
return err
}

Reply:
done